src/createAnimatedComponent/PropsFilter.tsx

```diff
@@ -37,7 +37,7 @@ export class PropsFilter implements IPropsFilter {
         const processedStyle: StyleProps = styles.map((style) => {
           if (style && style.viewDescriptors) {
             // this is how we recognize styles returned by useAnimatedStyle
-            style.viewsRef.add(component);
+            style.viewsRef?.add(component);
```

Comment:
Do we really need to make this change here?

Reply:
Because of inconsistency in `useAnimatedStyle`. We would always make `viewsRef` object in it, but also make `maybeViewsRef` object. The latter was sent to some functions while the former was always returned.

Therefore in native implementations we would unnecessarily return it and add to it in this place. Now that we return undefined instead we need to check if it's defined.